Hazel/src/Hazel/Renderer/Renderer2D.cpp

```diff
@@ -135,12 +135,15 @@ namespace Hazel {
 
 	void Renderer2D::Flush()
 	{
-		// Bind textures
-		for (uint32_t i = 0; i < s_Data.TextureSlotIndex; i++)
-			s_Data.TextureSlots[i]->Bind(i);
+		if (s_Data.QuadIndexCount > 0)
+		{
+			// Bind textures
+			for (uint32_t i = 0; i < s_Data.TextureSlotIndex; i++)
+				s_Data.TextureSlots[i]->Bind(i);
 		
-		RenderCommand::DrawIndexed(s_Data.QuadVertexArray, s_Data.QuadIndexCount);
-		s_Data.Stats.DrawCalls++;
+			RenderCommand::DrawIndexed(s_Data.QuadVertexArray, s_Data.QuadIndexCount);
+			s_Data.Stats.DrawCalls++;
+		}
```

Comment:
Since we decided to put the drawcall increment inside the if statement, it might be better to prematurely return instead? (github is messing up the suggestion, so i'm adding it as a regular code block instead)

```cpp
		if (s_Data.QuadIndexCount == 0)
			return; // Nothing to draw
		
		// Bind textures
		for (uint32_t i = 0; i < s_Data.TextureSlotIndex; i++)
			s_Data.TextureSlots[i]->Bind(i);
		
		RenderCommand::DrawIndexed(s_Data.QuadVertexArray, s_Data.QuadIndexCount);
		s_Data.Stats.DrawCalls++;
```